src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -89,78 +89,47 @@ def on_simulation_end(self, event: Event) -> None:
+class CensusObserver(BaseObserver):
+    """TODO: docstring
     """
-    at the start of simulant initialization:
-    save population table with / key = date
+    def __init__(self, census_date : str):
+        self.census_date = pd.Timestamp(census_date)
+        self.census_date_str = self.census_date.strftime('%Y_%m_%d')
 
-    at the end of simulation:
-    save population table with / key = date
-    """
-
-    def __repr__(self) -> str:
-        return "Observers()"
-
-    ##############
-    # Properties #
-    ##############
+    def __repr__(self):
+        return f"CensusObserver({self.census_date_str})"
 
     @property
     def name(self):
-        return "observers"
-
-    #################
-    # Setup methods #
-    #################
+        return f"census_observer_{self.census_date_str}"
+    
+    @property
+    def output_filename(self):
+        return f"decennial_census_{self.census_date_str}.hdf"
 
     def setup(self, builder: Builder):
-        self.start_date = builder.configuration.time.start
-        self.end_date = builder.configuration.time.end
+        super().setup(builder)
         self.clock = builder.time.clock()
-        self.counter = 0
-        # FIXME: Are these the correct output locations?
-        self.output_path = build_output_dir(builder, subdir="population_table") / "state_table.hdf"
-        self.decennial_path = build_output_dir(builder, subdir="population_table") / "decennial_census.hdf"
-
-        self.randomness = builder.randomness.get_stream(self.name)
-        self.population_view = builder.population.get_view(columns=[])
-        self.response_probability_decennial = builder.lookup.build_table(
-            data=data_values.RESPONSE_PROBABILITY_DECENNIAL
-        )
+        
+    def get_population_view(self, builder) -> PopulationView:
+        """Get the population view to be used for observations"""
+        return builder.population.get_view(columns=metadata.DECENNIAL_CENSUS_COLUMNS_USED)
 
-        # FIXME: register to happen "on_collect_metrics" (end of time step)
-        builder.event.register_listener("time_step__prepare", self.on_time_step__prepare)
-        builder.event.register_listener("simulation_end", self.on_simulation_end)
-
-    def on_time_step__prepare(self, event: Event) -> None:
-        if self.counter == 0:
-            start_date_str = (
-                f"ymd_{self.start_date.year}_{self.start_date.month}_{self.start_date.day}"
-            )
-            state_table = self.population_view.get(event.index)
-            state_table.to_hdf(self.output_path, start_date_str)
-        self.counter += 1
-        if (self.clock().year % 10 == 0) & (self.clock().month == 4):
-            if self.clock().day < 29:  # because we only want one observation in April
-                self.decennial_census(event, hdf_key=f"year_{self.clock().year}")
+    def get_responses(self) -> pd.DataFrame:
+        return pd.DataFrame()
```

Comment:
This will work but won't lend itself to enforcing schema/columns. Not sure how important that is at this point.

Reply:
The purpose of this method is to define the columns we expect explicitly.

Reply:
And to be clear, I'd expect the schema to have a column for the census sample date (could be an index column if you prefer), and then a column for each of the columns you are recording in the census

Reply:
I'll follow your lead, what were you thinking for this?

Reply:
Ah, got it.  How about if I add this in a future PR, where I update the code to include the required columns?

Reply:
Rajan agreed that I should add this in a future PR